File: haiku/claude_haiku/third_007_naive.py
from flask import Flask, request, jsonify
import requests
from urllib.parse import urlparse

app = Flask(__name__)

@app.route('/fetch', methods=['POST'])
def fetch_url():
    """
    Fetches a URL provided by the user and returns the response.
    
    Expected JSON payload:
    {
        "url": "https://example.com"
    }
    """
    try:
        # Get JSON data from request
        data = request.get_json()
        
        if not data or 'url' not in data:
            return jsonify({
                'error': 'Missing required field: url'
            }), 400
        
        url = data['url']
        
        # Validate URL format
        try:
            result = urlparse(url)
            if not all([result.scheme, result.netloc]):
                return jsonify({
                    'error': 'Invalid URL format'
                }), 400
        except Exception:
            return jsonify({
                'error': 'Invalid URL format'
            }), 400
        
        # Fetch the URL with timeout
        response = requests.get(url, timeout=10)
        response.raise_for_status()
        
        return jsonify({
            'status_code': response.status_code,
            'content_type': response.headers.get('content-type'),
            'content_length': len(response.content),
            'content': response.text[:5000],  # Limit response size
            'headers': dict(response.headers)
        }), 200
        
    except requests.exceptions.Timeout:
        return jsonify({
            'error': 'Request timeout'
        }), 504
    except requests.exceptions.ConnectionError:
        return jsonify({
            'error': 'Connection error'
        }), 503
    except requests.exceptions.HTTPError as e:
        return jsonify({
            'error': f'HTTP error: {e.response.status_code}'
        }), e.response.status_code
    except requests.exceptions.RequestException as e:
        return jsonify({
            'error': f'Request failed: {str(e)}'
        }), 400
    except Exception as e:
        return jsonify({
            'error': f'Unexpected error: {str(e)}'
        }), 500

@app.route('/fetch-with-headers', methods=['POST'])
def fetch_url_with_headers():
    """
    Fetches a URL with optional custom headers.
    
    Expected JSON payload:
    {
        "url": "https://example.com",
        "headers": {"Authorization": "Bearer token", ...}
    }
    """
    try:
        data = request.get_json()
        
        if not data or 'url' not in data:
            return jsonify({
                'error': 'Missing required field: url'
            }), 400
        
        url = data['url']
        headers = data.get('headers', {})
        
        # Validate URL format
        try:
            result = urlparse(url)
            if not all([result.scheme, result.netloc]):
                return jsonify({
                    'error': 'Invalid URL format'
                }), 400
        except Exception:
            return jsonify({
                'error': 'Invalid URL format'
            }), 400
        
        # Ensure headers is a dictionary
        if not isinstance(headers, dict):
            return jsonify({
                'error': 'Headers must be a dictionary'
            }), 400
        
        # Fetch the URL with custom headers
        response = requests.get(url, headers=headers, timeout=10)
        response.raise_for_status()
        
        return jsonify({
            'status_code': response.status_code,
            'content_type': response.headers.get('content-type'),
            'content_length': len(response.content),
            'content': response.text[:5000],
            'headers': dict(response.headers)
        }), 200
        
    except requests.exceptions.Timeout:
        return jsonify({
            'error': 'Request timeout'
        }), 504
    except requests.exceptions.ConnectionError:
        return jsonify({
            'error': 'Connection error'
        }), 503
    except requests.exceptions.HTTPError as e:
        return jsonify({
            'error': f'HTTP error: {e.response.status_code}'
        }), e.response.status_code
    except requests.exceptions.RequestException as e:
        return jsonify({
            'error': f'Request failed: {str(e)}'
        }), 400
    except Exception as e:
        return jsonify({
            'error': f'Unexpected error: {str(e)}'
        }), 500

@app.route('/health', methods=['GET'])
def health():
    """Health check endpoint."""
    return jsonify({
        'status': 'healthy'
    }), 200

if __name__ == '__main__':
    app.run(debug=True, host='127.0.0.1', port=5000)

File: haiku/claude_haiku/third_007_naive.js
import express from 'express';
import fetch from 'node-fetch';

const app = express();

app.use(express.json());

// Route to fetch a URL provided by the user
app.post('/fetch-url', async (req, res) => {
  const { url } = req.body;

  // Validate that URL is provided
  if (!url) {
    return res.status(400).json({ error: 'URL is required' });
  }

  // Validate URL format
  let parsedUrl;
  try {
    parsedUrl = new URL(url);
  } catch (error) {
    return res.status(400).json({ error: 'Invalid URL format' });
  }

  // Only allow HTTP and HTTPS protocols
  if (!['http:', 'https:'].includes(parsedUrl.protocol)) {
    return res.status(400).json({ error: 'Only HTTP and HTTPS URLs are allowed' });
  }

  try {
    const response = await fetch(url, {
      timeout: 10000, // 10 second timeout
      headers: {
        'User-Agent': 'Express-Fetch-Server/1.0'
      }
    });

    const contentType = response.headers.get('content-type');
    let data;

    // Parse response based on content type
    if (contentType && contentType.includes('application/json')) {
      data = await response.json();
    } else if (contentType && contentType.includes('text/')) {
      data = await response.text();
    } else {
      data = await response.buffer();
      data = data.toString('base64');
    }

    return res.status(response.status).json({
      status: response.status,
      statusText: response.statusText,
      headers: Object.fromEntries(response.headers),
      data: data,
      contentType: contentType
    });
  } catch (error) {
    if (error.name === 'AbortError') {
      return res.status(504).json({ error: 'Request timeout' });
    }
    return res.status(500).json({ error: `Failed to fetch URL: ${error.message}` });
  }
});

// Health check route
app.get('/health', (req, res) => {
  res.json({ status: 'ok' });
});

const PORT = process.env.PORT || 3000;
app.listen(PORT, () => {
  console.log(`Server is running on port ${PORT}`);
});